### ISSUE TITLE
Tag internal metrics with event-level data

### ISSUE THREAD

**Issue by binarylogic** [vectordotdev/vector#9257]
To drive useful insights around Vector's activity, we need to tag internal metrics with event-level data. This is largely dependent on #9256 which, in turn, is dependent on the upcoming schema work.

For example, the ability to tag Vector's internal metrics with the origin service name allows users to facet internal metrics along service lines. With this users could see how much data a single service is sending to Vector and how much of that service's data fans out across sinks.

**Comment by spencergilbert** [comment]
Would this be a part of the `internal_metrics` source config?

**Comment by binarylogic** [comment]
No, I imagine it would need to be updated in Vector's topology as context that gets added to metrics. I'm sure there is a better way to do this for performance reasons, but the end result would be the same.

**Comment by spencergilbert** [comment]
I wonder if there would be times those tags wouldn't be desireable for cardinatlity reasons

**Comment by jszwedko** [comment]
I think this is a duplicate of https://github.com/vectordotdev/vector/issues/7403

**Comment by binarylogic** [comment]
Closing in favor of #7403
